test/e2e/tests/eth-sign.spec.js

```diff
@@ -101,4 +101,87 @@ describe('Eth sign', function () {
       },
     );
   });
+
+  it('can queue multiple eth sign and confirm', async function () {
+    const expectedPersonalMessage =
+      '0x879a053d4800c6354e76c7985a865d2922c82fb5b3f4577b2fe08b998954f2e0';
+    const expectedEthSignResult =
+      '"0x816ab6c5d5356548cc4e004ef35a37fdfab916742a2bbeda756cd064c3d3789a6557d41d49549be1de249e1937a8d048996dfcc70d0552111605dc7cc471e8531b"';
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPreferencesController({
+            disabledRpcMethodPreferences: {
+              eth_sign: true,
+            },
+          })
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await openDapp(driver);
+        // Create eth sign
+        await driver.clickElement('#ethSign');
+
+        // Wait for Signature request popup
+        await driver.waitUntilXWindowHandles(3);
+        let windowHandles = await driver.getAllWindowHandles();
+
+        // Switch to Dapp
+        await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
+
+        // Create second eth sign
+        await driver.clickElement('#ethSign');
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+
+        await driver.waitForSelector({
+          text: 'Reject 2 requests',
+          tag: 'a',
+        });
+
+        const title = await driver.findElement(
+          '.request-signature__content__title',
+        );
+        const origin = await driver.findElement('.request-signature__origin');
+        assert.equal(await title.getText(), 'Signature request');
```

Comment:
Is it better to use non-text selectors to prevent them being too brittle, and also support other locales?